core/templates/domain/statistics/playthrough-issue.model.ts

```diff
@@ -102,59 +66,54 @@ abstract class PlaythroughIssueBase<PlaythroughIssueType> {
       is_valid: this.isValid,
     };
   }
-}
-
-export class EarlyQuitPlaythroughIssue extends
-  PlaythroughIssueBase<PlaythroughIssueType.EarlyQuit> {
-  getStateNameWithIssue(): string {
-    return this.issueCustomizationArgs.state_name.value;
-  }
-}
 
-export class MultipleIncorrectSubmissionsPlaythroughIssue extends
-  PlaythroughIssueBase<PlaythroughIssueType.MultipleIncorrectSubmissions> {
   getStateNameWithIssue(): string {
-    return this.issueCustomizationArgs.state_name.value;
-  }
-}
-
-export class CyclicStateTransitionsPlaythroughIssue extends
-  PlaythroughIssueBase<PlaythroughIssueType.CyclicStateTransitions> {
-  getStateNameWithIssue(): string {
-    const stateNames = this.issueCustomizationArgs.state_names.value;
-    return stateNames[stateNames.length - 1];
+    switch (this.issueType) {
+      case PlaythroughIssueType.EarlyQuit: {
+        const args = this.issueCustomizationArgs as EarlyQuitCustomizationArgs;
+        return args.state_name.value;
+      }
+      case PlaythroughIssueType.MultipleIncorrectSubmissions: {
+        const args = this
+          .issueCustomizationArgs as
+            MultipleIncorrectSubmissionsCustomizationArgs;
+        return args.state_name.value;
+      }
+      case PlaythroughIssueType.CyclicStateTransitions: {
+        const args = this
+          .issueCustomizationArgs as CyclicStateTransitionsCustomizationArgs;
+        const stateNames = args.state_names.value;
+        return stateNames[stateNames.length - 1];
+      }
+    }
```

Comment:
Should there be a default specified here and in other switch/cases?

Reply:
Two cases we need to check `CustomizationArgs`: 
- creating a new instance of class `createFromBackendDict` and 
- getting info about an existing instance `getStateNameWithIssue`.

Instance cannot exist with wrong `issueType` TypeScript will protect against such situation. 
So for `getStateNameWithIssue` `default` keyword in the `switch` statement are redundant.

We can through error or add return, but then testing is mandatory. 
Honestly, I don't know how to create instance of class with wrong `this.issueType` for testing `default:`.

I added comment for easier code understanding, less thinking.
```
// Instance cannot exist with wrong issueType
// default: Never happens.
```